### PR TITLE
MEN-4671: Add test_update_module

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

File: tests/test_update_modules.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        file_tree = tempfile.mkdtemp()
+        try:
+            files_and_content = ["file1", "file2"]
+            for file_and_content in files_and_content:
+                with open(os.path.join(file_tree, file_and_content), "w") as fd:
+                    fd.write(file_and_content)
+
+            artifact_file = os.path.join(file_tree, "update.mender")
+            cmd = (
+                "directory-artifact-gen -o %s -n update-directory -t qemux86-64 -d /tmp/test_directory_update_module %s"
```
Comment (kacf) [inline tests/test_update_modules.py:44]: `qemux86-64` needs to be replaced with `bitbake_variables["MACHINE"]`, and you need to include `bitbake_variables` in the test arguments to import the fixture. I would expect this to fail on vexpress-qemu without it.

There is one more below, make sure you get both.

File: tests/test_update_modules.py [hunk 2]
```diff
@@ -74,3 +74,84 @@ def test_directory_update_module(self, connection):
...
+        try:
+            update_file = os.path.join(file_tree, "my-file")
+            artifact_file = os.path.join(file_tree, "update.mender")
+
+            # Create and send first update
+            with open(update_file, "w") as fd:
+                fd.write("my-initial-content")
+            os.chmod(update_file, 0o777)
+            cmd = (
+                "single-file-artifact-gen -o %s -n update-file-v1 -t qemux86-64 -d /tmp/some/new/path %s"
```
Comment (kacf) [inline tests/test_update_modules.py:141]: Same here, ref qemux86-64.

Again, make sure you get both.

**Review by kacf — APPROVED** [review]
Nice tests! Pre-approving, the comment below is important, but straightforward to fix.

**Comment by lluiscampos** [comment]
@kacf Thanks.

I fixed the hardcoded device-type and also added the new future releases of Yocto to the fixture (last commit).

**Review by kacf — APPROVED** [review]

**Pull request merged**